js/directives/updateBackground.js

```diff
@@ -34,36 +34,24 @@ var myUpdateBackgroundFactory = function(appSettings, $http) {
           .css('background', background);
     };
 
-    var requestThumbnail = function(element, url, tries) {
+    var requestThumbnail = function(element, url) {
 
         return $http({
-            url: 'http://api.page2images.com/restfullink?p2i_url=' +
-            encodeURIComponent(url) +
-            '&p2i_device=6&p2i_size=400x150&p2i_screen=1024x768&p2i_imageformat=jpg&p2i_wait=0&p2i_key=7cd903b37d087238',
-            method: "POST",
+            url: 'http://dewey-server.azurewebsites.net/screenshot?query=' +
+            encodeURIComponent(url),
+            method: "GET",
             headers: {
                 'Content-Type': 'application/json; charset=utf-8'
             }
         }).success(function(responseData) {
-            if (responseData.status === "processing") {
-                // Server processing results, let's try later in 3 seconds,
-                // but not more than 40 times (120 seconds)
-                if (tries <= 40) {
-                    setTimeout(function() {
-                        requestThumbnail(element, url, ++tries);
-                    }, ((parseInt(responseData.estimated_need_time) || 2) + 1) * 1000);
-                }
-            } else if(responseData.status === "finished" && responseData.image_url){
-                // Finished, let's update thumbnail
-                updateThumbnail(element, responseData.image_url);
-            }
+            updateThumbnail(element, responseData.image_url);
```

Comment:
does it mean that your server handles these retries?
does it block http call till it can get result?

If yes I don't think that this will work. Browser is limited to 2-6 connections. If you will block all connections with these requests - you will not be able to start new requests.